src/test/javascript/spec/component/course/course-detail.component.spec.ts

```diff
@@ -1,32 +1,151 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
 import { ActivatedRoute } from '@angular/router';
 import { of } from 'rxjs';
 import { ArtemisTestModule } from '../../test.module';
 import { CourseDetailComponent } from 'app/course/manage/course-detail.component';
-import { Course } from 'app/entities/course.model';
 import { TranslateService } from '@ngx-translate/core';
-import { MockProvider } from 'ng-mocks';
+import { MockComponent, MockProvider } from 'ng-mocks';
+import { SecuredImageComponent } from 'app/shared/image/secured-image.component';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import * as moment from 'moment';
+import { MockActivatedRoute } from '../../helpers/mocks/activated-route/mock-activated-route';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import * as sinon from 'sinon';
+import { HttpResponse } from '@angular/common/http';
+import { JhiAlertService } from 'ng-jhipster';
 
 describe('Course Management Detail Component', () => {
     let comp: CourseDetailComponent;
     let fixture: ComponentFixture<CourseDetailComponent>;
-
-    const course: Course = { id: 123 } as Course;
-    const route = ({ data: of({ courseId: course.id }), children: [] } as any) as ActivatedRoute;
+    let courseManagementService: CourseManagementService;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
             imports: [ArtemisTestModule],
-            declarations: [CourseDetailComponent],
-            providers: [{ provide: ActivatedRoute, useValue: route }, MockProvider(TranslateService)],
+            declarations: [CourseDetailComponent, MockComponent(SecuredImageComponent)],
+            providers: [
+                { provide: ActivatedRoute, useValue: new MockActivatedRoute() },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+                MockProvider(JhiAlertService),
+            ],
         })
             .overrideTemplate(CourseDetailComponent, '')
```

Comment:
If you are bored you could remove this `overrideTemplate` since we don't want to use this anymore :P